api-operator/pkg/envoy/server/api/restserver/configure_restapi.go

```diff
@@ -0,0 +1,174 @@
+// This file is safe to edit. Once it exists it will not be overwritten
+
+package restserver
+
+import (
+	"crypto/tls"
+	"github.com/go-openapi/errors"
+	"github.com/go-openapi/loads"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/jessevdk/go-flags"
+	"github.com/wso2/k8s-api-operator/api-operator/pkg/envoy/server/api/models"
+	"github.com/wso2/k8s-api-operator/api-operator/pkg/envoy/server/api/restserver/operations"
+	"github.com/wso2/k8s-api-operator/api-operator/pkg/envoy/server/api/restserver/operations/a_p_is_all"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var logRestServer = logf.Log.WithName("server.operator")
+
+//go:generate swagger generate server --target ../../api --name Restapi --spec ../../resources/apiSwagger.yaml --server-package restserver --principal models.Principal
+
+func configureFlags(api *operations.RestapiAPI) {
+	// api.CommandLineOptionsGroups = []swag.CommandLineOptionsGroup{ ... }
+}
+
+func configureAPI(api *operations.RestapiAPI) http.Handler {
+	// configure the api here
+	api.ServeError = errors.ServeError
+
+	// Set your custom logger if needed. Default one is log.Printf
+	// Expected interface func(string, ...interface{})
+	//
+	// Example:
+	// api.Logger = log.Printf
+
+	api.UseSwaggerUI()
+	// To continue using redoc as your UI, uncomment the following line
+	// api.UseRedoc()
+
+	api.JSONConsumer = runtime.JSONConsumer()
+
+	api.ApplicationZipProducer = runtime.ProducerFunc(func(w io.Writer, data interface{}) error {
+		dataZip, err := ioutil.ReadFile(os.TempDir() + "/done.zip")
+		if err != nil {
+			logRestServer.Error(err, "Error when reading zipped file")
+			return err
+		}
+		if dataZip != nil {
+			_, err = w.Write(dataZip)
+			if err != nil {
+				logRestServer.Error(err, "Error when writing zipped file to response")
+				return err
+			}
+		}
+		return nil
+	})
+
+	// Applies when the Authorization header is set with the Basic scheme
+	api.BasicAuthAuth = func(user string, pass string) (*models.Principal, error) {
+		if user != "admin" || pass != "admin" {
+			return nil, errors.New(401, "Credentials are invalid")
+		}
+
+		p := models.Principal{
+			Token:    "xxxx",
+			Tenant:   "xxxx",
+			Username: user,
+		}
+		return &p, nil
+	}
+
+	// Set your custom authorizer if needed. Default one is security.Authorized()
+	// Expected interface runtime.Authorizer
+	//
+	// Example:
+	// api.APIAuthorizer = security.Authorized()
+	if api.ApIsAllGetApisHandler == nil {
+		api.ApIsAllGetApisHandler = a_p_is_all.GetApisHandlerFunc(func(params a_p_is_all.GetApisParams, principal *models.Principal) middleware.Responder {
```

Comment:
Length exceeded(120 characters). Shall we break into two lines?

Reply:
Fixed with 28191756937ef7cf2473f7531e68b5a81ae87e68